File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    answer = 0
    n, m = len(board), len(board[0])
    directions = [(-1, 0), (1, 0), (0, -1), (0, 1)]
    
    color = board[h][w]
    for d in directions:
        nh, nw = h + d[0], w + d[1]
        if 0 <= nh < n and 0 <= nw < m:
            if board[nh][nw] == color:
                answer += 1
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        int n = board.length; // 보드판 크기
        
        int[][] directions = new int[][]{ {-1, 0}, {1, 0}, {0, -1}, {0, 1} };
        
        String color = board[h][w];
        
        for (int[] dir: directions) {
            int nh = h + dir[0], nw = w + dir[1];
            if (nh >= 0 && nh < n && nw >= 0 && nw < n) {
                if (board[nh][nw].equals(color)) answer++;
            }
        }
        
        return answer;
    }
}